src/isEmpty.ts

```diff
@@ -48,5 +48,10 @@ export function isEmpty(data: object | string | undefined): boolean {
     return data.length === 0;
   }
 
-  return Object.keys(data).length === 0;
+  for (const key in data) {
+    if (Object.hasOwn(data, key)) {
+      return false;
+    }
+  }
```

Comment:
**question:** does using `of` and then omitting the `hasOwn` work?

Reply:
Objects aren't iterable by default, so forof won't work on them. For forin, it will loop over the inherited properties as well, so an empty object with a non-empty prototype chain would return not empty unless the hasOwn check is done.

Reply:
that makes sense